test/engine/SpatialJoinTest.cpp

```diff
@@ -0,0 +1,1957 @@
+// case call the function orderColAccordingToVarColMap
+std::vector<std::string> createRowVectorFromColumnVector(
+    std::vector<std::vector<std::string>> column_vector) {
+  std::vector<std::string> result;
+  if (column_vector.size() > 0) {
+    for (size_t i = 0; i < column_vector.at(0).size(); i++) {
+      std::string str = "";
+      for (size_t k = 0; k < column_vector.size(); k++) {
+        str += column_vector.at(k).at(i);
+        str += " ";
+      }
+      result.push_back(str.substr(0, str.size() - 1));
+    }
+  }
+  return result;
+}
+
+// this function compares the ResultTable resultTableToTest with the
+// expected_output. It assumes, that all rows are unique. Therefore the tables
+// are equal, if each row of the expected_output is contained in the table and
+// the tables have the same size
+void compareResultTable(  // const QueryExecutionContext* qec,
+                          // const std::shared_ptr<const ResultTable>
+                          // resultTableToTest,
+    std::vector<std::string> tableToTest,
+    std::vector<std::string>* expected_output) {
+  // rows will be a reordered version of the tableToTest
+  std::vector<std::string> rows(expected_output->size(), "");
+  // std::vector<std::string> tableToTest = printTable(qec, resultTableToTest);
+  for (size_t i = 0; i < expected_output->size(); i++) {
+    rows.at(i) = tableToTest.at(i);  // initialize as copy, reorder when needed
+  }
+  for (size_t i = 0; i < expected_output->size(); i++) {
+    for (size_t k = 0; k < tableToTest.size(); k++) {
+      if (tableToTest.at(k) == expected_output->at(i)) {
+        rows.at(i) = tableToTest.at(k);  // change order of expected output
+        break;
+      }
+    }
+  }
+  // if all rows from the expected output are in the result table and the tables
+  // are equally large, then both tables are equal (assuming each row is unique)
+  ASSERT_EQ(tableToTest.size(), expected_output->size());
+  for (size_t i = 0; i < expected_output->size(); i++) {
+    ASSERT_EQ(expected_output->at(i), rows.at(i));
+  }
+}
+
+// create a small test dataset, which focuses on points as geometry objects
+std::string createSmallDatasetWithPoints() {
+  // note, that some of these objects have a polygon representation, but for
+  // testing purposes, they get represented as a point here. I took those
+  // points, such that it is obvious, which pair of objects should be included,
+  // when the maximum distance is x meters. Please note, that these datapoints
+  // are not copied from a real input file. Copying the query will therefore
+  // likely not result in the same results as here (also the names, coordinates,
+  // etc. might be different in the real datasets)
+
+  // Uni Freiburg TF, building 101
+  std::string kg = "<node_1> <name> \"Uni Freiburg TF\" .";
+  kg += "<node_1> <hasGeometry> <geometry1> .";
+  kg += "<geometry1> <asWKT> \"POINT(7.83505 48.01267)\" .";
+  // Minster Freiburg
```

Comment:
Get rid of lines by a 
`addNode(name, geometry)` lambda that gets rid of your repetitive boilerplate.